server/src/main/java/org/elasticsearch/index/snapshots/blobstore/SlicedInputStream.java

```diff
@@ -148,11 +166,15 @@ public void reset() throws IOException {
                     throw new IOException("Mark has not been set");
                 }
 
-                // We do not call the SlicedInputStream's skipNBytes but call skipNBytes directly on the returned stream, to ensure that
-                // the skip is performed on the marked slice and no other slices are involved. This may help uncover any bugs.
                 nextSlice = markedSlice;
+                if (currentStream == null) {
+                    // In case EOF has been reached, we set the currentStream to a non-null value so that nextStream() does not complain.
+                    currentStream = InputStream.nullInputStream();
+                }
```

Comment:
Do we really need to do this dance just to make `nextStream()` happy? It seems as if we could just set `currentStream` to the right slice directly?

Reply:
Your comment made me thinking and found another non-intrusive way: extend the assertion in nextStream(). Feel free to review!